docs/parser.md

```diff
@@ -0,0 +1,502 @@
+| `button`      | `ContentButtonComponent`      | `src/app/page/component/content-button/`      |
+| `link`        | `ContentLinkComponent`        | `src/app/page/component/content-link/`        |
+| `card`        | `ContentCardComponent`        | `src/app/page/component/content-card/`        |
+| `video`       | `ContentVideoComponent`       | `src/app/page/component/content-video/`       |
+| `form`        | `ContentFormComponent`        | `src/app/page/component/content-form/`        |
+| `input`       | `ContentInputComponent`       | `src/app/page/component/content-input/`       |
+| `accordion`   | `ContentAccordionComponent`   | `src/app/page/component/content-accordion/`   |
+| `tabs`        | `ContentTabsComponent`        | `src/app/page/component/content-tabs/`        |
+| `animation`   | `ContentAnimationComponent`   | `src/app/page/component/content-animation/`   |
+| `multiselect` | `ContentMultiselectComponent` | `src/app/page/component/content-multiselect/` |
+| `flow`        | `ContentFlowComponent`        | `src/app/page/component/content-flow/`        |
+| `spacer`      | `ContentSpacerComponent`      | `src/app/page/component/content-spacer/`      |
+
+### Full Flow Example
+
+```
+XML:
+<page>
+  <content:paragraph>
+    <content:text text-color="#333333">Hello World</content:text>
+  </content:paragraph>
+</page>
+
+↓ (Kotlin Parser)
+
+TypeScript Object:
+Paragraph {
+  content: [
+    Text {
+      text: "Hello World",
+      textColor: Color(51, 51, 51)
+    }
+  ]
+}
+
+↓ (ContentRepeater)
+
+ContentParagraphComponent receives Paragraph
+  ↓
+ContentTextComponent receives Text
+  ↓
+Renders: <span style="color: #333333">Hello World</span>
+```
+
+## Event System
+
+The event system enables interactive navigation and actions within GodTools content.
+
+### EventId Structure
+
+Events are identified by an `EventId` object:
+
+```typescript
+class EventId {
+  namespace: string | null; // Optional namespace
+  name: string; // Event name
+}
+```
+
+Events can be specified in XML as:
+
+- `name` - Simple event (e.g., `next-page`)
+- `namespace:name` - Namespaced event (e.g., `state:selectedOption`)
+
+**Kotlin Definition**: `kotlin-mpp-godtools-tool-parser/module/parser-base/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/EventId.kt`
+
+### Special Event Namespaces
+
+#### `state:` Namespace
+
+Events with the `state:` namespace are **dynamic** and resolved at runtime from the parser State:
+
+```typescript
+// XML: events="state:selectedOption"
+// If State has: selectedOption = ["option-a", "option-b"]
+// Then resolve() returns: [EventId("option-a"), EventId("option-b")]
+
+eventId.resolve(state);
+```
+
+This allows content to trigger different navigation based on user choices stored in State.
+
+**Kotlin Implementation**: `kotlin-mpp-godtools-tool-parser/module/renderer-state/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/EventId+State.kt`
```

Comment:
this code reference is outdated due to several refactoring passes to support the new CMP renderer and still expose this to JS.

https://github.com/CruGlobal/kotlin-mpp-godtools-tool-parser/blob/main/module/parser/src/commonMain/kotlin/org/cru/godtools/shared/tool/parser/model/EventId.kt#L31-L34